torch/csrc/cuda/GdsFile.cpp

```diff
@@ -0,0 +1,131 @@
+#include <pybind11/pybind11.h>
+#include <torch/csrc/utils/pybind.h>
+
+#if !defined(USE_ROCM) && defined(USE_CUFILE)
```

Comment:
The USE_ROCM is not needed anymore?
I would expect that setting both USE_CUFILE and USE_ROCM should raise at the cmake level?